src/docfx/build/link/LinkResolver.cs

```diff
@@ -145,6 +145,13 @@ internal class LinkResolver
                 return (Errors.Link.LinkOutOfScope(href, file), href, fragment, linkType, null, false);
             }
 
+            // For static hosting, reference file in fallback repo should be resolved to docs site URL
+            if (file.FilePath.Origin == FileOrigin.Fallback && file.ContentType != ContentType.Resource && _config.OutputUrlType != OutputUrlType.Docs)
```

Comment:
consider be more specific here
```suggestion
            if (file.FilePath.Origin == FileOrigin.Fallback && file.ContentType == ContentType.Page && _config.OutputUrlType != OutputUrlType.Docs)
```